packages/fxa-settings/src/lib/error-utils.ts

```diff
@@ -0,0 +1,175 @@
+
+import { GraphQLError } from 'graphql';
+import {
+  AuthUiError,
+  AuthUiErrorNos,
+  AuthUiErrors,
+} from './auth-errors/auth-errors';
+import VerificationMethods from '../constants/verification-methods';
+import VerificationReasons from '../constants/verification-reasons';
+import * as Sentry from '@sentry/browser';
+import { FtlMsgResolver } from 'fxa-react/lib/utils';
+import { OAUTH_ERRORS } from './oauth';
+
+// TODO: Consolidate with AuthUiError type
+export interface GenericError {
+  errno: number;
+  message: string;
+  /** Not typically used in the UI, prefer 'errno' instead */
+  code?: number;
+  retryAfter?: number;
+  retryAfterLocalized?: string;
+}
+
+export type BeginSigninError = GenericError & {
+  verificationReason?: VerificationReasons;
+  verificationMethod?: VerificationMethods;
+  /** Important for 'Incorrect email case' edgecase */
+  email?: string;
+};
+
+export type HandledError = GenericError | BeginSigninError;
+
+const handleAuthUiError = (error: {
+  errno: number;
+  message: string;
+}): { error: HandledError } => {
+  const { errno } = error as HandledError;
+  if (errno && AuthUiErrorNos[errno]) {
+    return { error };
+  }
+  return { error: AuthUiErrors.UNEXPECTED_ERROR as HandledError };
+};
+
+export const getHandledError = (error: {
+  errno: number;
+  message: string;
+  graphQLErrors?: GraphQLError[];
+}) => {
+  const graphQLError: GraphQLError | undefined = error.graphQLErrors?.[0];
+  if (graphQLError) {
+    return handleGQLError(graphQLError);
+  }
+  return handleAuthUiError(error);
+};
+
+const handleGQLError = (graphQLError: GraphQLError) => {
+  const errno = graphQLError.extensions.errno as number;
+
+  if (errno && AuthUiErrorNos[errno]) {
+    const uiError = {
+      message: AuthUiErrorNos[errno].message,
+      errno,
+      email:
+        errno === AuthUiErrors.INCORRECT_EMAIL_CASE.errno
+          ? graphQLError.extensions.email
+          : undefined,
+      verificationMethod:
+        (graphQLError.extensions.verificationMethod as VerificationMethods) ||
+        undefined,
+      verificationReason:
+        (graphQLError.extensions.verificationReason as VerificationReasons) ||
+        undefined,
+      retryAfter: (graphQLError.extensions.retryAfter as number) || undefined,
+      retryAfterLocalized:
+        (graphQLError?.extensions.retryAfterLocalized as string) || undefined,
+    };
+    return { error: uiError as HandledError };
+  }
+
+  // if not a graphQLError or if no localizable message available for error
+  return { error: AuthUiErrors.UNEXPECTED_ERROR as HandledError };
+};
+
+/**
+ * Utility function to retrieve the localized auth client error message
+ * - works for throttling errors that include a localized retry after value
+ * - returns an unexpected error if the error is unknown or does not have a localized message
+ * @param ftlMsgResolver
+ * @param err is an AuthClient error
+ * @returns
+ */
+export const getLocalizedErrorMessage = (
```

Comment:
I moved this here and out from `auth-errors` since we're using it for oauth errors as well.